packages/dds/map/src/interfaces.ts

```diff
@@ -346,11 +348,14 @@ export interface ISharedMapEvents extends ISharedObjectEvents {
  *
  * For more information, including example usages, see {@link https://fluidframework.com/docs/data-structures/map/}.
  * @sealed
- * @alpha
+ * @beta
  */
-// TODO: Use `unknown` instead (breaking change).
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export interface ISharedMap extends ISharedObject<ISharedMapEvents>, Map<string, any> {
+export interface ISharedMap
+	extends IEventProvider<ISharedMapEvents>,
```

Comment:
This is the API for map used in the declarative API. Therefore, like with SharedTree, it can't pull in ISharedObject which transitively references many encapsulated API types, like IChannel, IFluidDataStoreRuntime etc.

Reply:
Making separate ISharedMap like interfaces for the imperative and declarative API surfaces would be another option, which might work out cleaner. Input on this is desired.

Reply:
This is the main thing I'm looking for feedback on. Would it be better to do what I did here, and modify the existing ISharedMap to remove these dependencies, or introduce a new ISharedMap specific to fluid-framework without them?

Reply:
It's not declarative and imperative, its declarative and encapsulated (I don't know why we use that name, but that's what we use).

In this case the declarative API is a subset, however the declarative API is located in fluid-framework, so having the encapsulated on depend on the declarative on is the opposite of our dependency layering. We could declare both in the Map package, but then they need different names. I suppose declaring both there, then reexporting the declarative one under a different name from fluid-framework would work, but that makes the docs a bit messy.

Anyway, thanks for the thoughts, I'll make a draft trying it out.

Reply:
(I was just trying to match "imperative" that you stated before. ;) )
There is a difference between interfaces and implementation. If declarative is strict subset interface-wise, that doesn't mean encapsulated depends on declarative for implementation.
I can envision that there is some amount of declarative support that is not a subset of imperative. Like some builder patterns that have to happen early in an initialization process. When that is the case, I think it is probably wise not to name something "declarative" just because that is properly identified subset. That will help us avoid automatically assuming that "declarative" is always okay to depend on, since some declarative use has baggage (extra code support).

Reply:
The packages which are the entry points for the declarative API import the packages which are the entry points for the encapsulated API and the code implementing the encapsulated API is an implementation detail of the declarative API. Thats what I mean when I say there is a dependency.

This makes is rather unclear how to organize things. Either the declarative API works like a wrapper where it abstracts away things, or we need all the code to know about both abstractions and support both. Right now we have a mix of both patterns.

Anyway, I made https://github.com/microsoft/FluidFramework/pull/21343 taking the wrapper approach, and it worked out pretty clean.